### PR TITLE
fix: problems with internet connection detection SQSERVICES-1917

### PR DESCRIPTION
**Pull request by MarcinR**
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability

### CODE REVIEW

File: wire-ios-sync-engine/Source/SessionManager/BackendEnvironmentProvider+Reachability.swift [hunk 1]
```diff
@@ -83,6 +83,7 @@ final class ReachabilityWrapper: NSObject, ReachabilityProvider, TearDownCapable
         didSet {
             if safeReachability == nil && enabled {
                 safeReachability = reachabilityClosure()
+                NotificationCenter.default.post(name: NSNotification.Name(rawValue: ZMTransportSessionReachabilityIsEnabled), object: nil)
```
Comment (netbe) [inline wire-ios-sync-engine/Source/SessionManager/BackendEnvironmentProvider+Reachability.swift]:
```suggestion
                NotificationCenter.default.post(name: NSNotification.Name(rawValue: ZMTransportSessionReachabilityIsEnabled), object: self)
```
We could limit to check only the ZMTransportSession that depends on this ReachabilityWrapper renews its observations

File: wire-ios-transport/Source/TransportSession/ZMTransportSession.m [hunk 2]
```diff
@@ -289,6 +290,7 @@ - (instancetype)initWithURLSessionsDirectory:(id<URLSessionsDirectory, TearDownC
         }];
 
         self.remoteMonitoring = [[RemoteMonitoring alloc] initWithLevel: LevelInfo];
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(renewReachabilityObserverToken) name:ZMTransportSessionReachabilityIsEnabled object:nil];
```
Comment (netbe) [inline wire-ios-transport/Source/TransportSession/ZMTransportSession.m]:
```suggestion
        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(renewReachabilityObserverToken) name:ZMTransportSessionReachabilityIsEnabled object:self.reachability];
```

**Review by netbe — CHANGES_REQUESTED** [review]
can you add a unit test to it too?

**Review by netbe — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]
Could you please add a description of the issue, the cause, and the solution? It’s helpful for reviewers to understand the context.

**Pull request merged**